Wodes_Grimoire/cards/Ensnare.ts

```diff
@@ -25,57 +23,33 @@ const spell: Spell = {
         expenseScaling: 1.5,
         probability: probabilityMap[CardRarity.SPECIAL], 
         thumbnail: 'spellmasons-mods/Wodes_grimoire/graphics/icons/spelliconEnsnare.png', //TODO
-        //animationPath,
         sfx: '', //TODO
         description: [`Prevents the target from moving. Furthur casts increase duration.`],
         effect: async (state, card, quantity, underworld, prediction) => {
-            //Only filter unit thats are alive and can attack. u.unitSubType[3] presumed SUPPORT_CLASS, figure out later even needed
+            //Only filter unit thats are alive.
             const targets = state.targetedUnits.filter(u => u.alive);
-            //Play for client
-            if (targets.length) {
-                if (!prediction && !globalThis.headless) {
-                    setTimeout(() => {
-                        playDefaultSpellSFX(card, prediction);
-                        for (let unit of targets) {
-                            //TODO: check if oneOffImage is even right, took from undead blade. but addOneOffAnimation might fit better
-                            //const spellEffectImage = oneOffImage(unit, animationPath, containerSpells);
-                            Unit.addModifier(unit, card.id, underworld, prediction, quantity);
-                        }
-                    }, 100) //TODO when animation is determined
-                } else {
-                    for (let unit of targets) {
-                        Unit.addModifier(unit, card.id, underworld, prediction, quantity);
-                    }
-                }
-            }
             //Refund if targets no one that can attack
             if (targets.length == 0) {
                 refundLastSpell(state, prediction, 'No target, mana refunded')
+            } else {
+                if (!prediction){
+                    playDefaultSpellSFX(card, prediction);
+                }
+                for (let unit of targets) {
+                    Unit.addModifier(unit, card.id, underworld, prediction, quantity);
+                }
             }
             if (!prediction && !globalThis.headless) {
                 await new Promise((resolve) => {
-                    setTimeout(resolve, 100); //Very unfamiliar with promises and async functions. 
+                    setTimeout(resolve, 100);
                 })
             }
```

Comment:
This whole block is unnecessary.  Promises and async are just ways of making something happen "later" rather than one line after another during code execution.

This block essentially says, "wait 100 milliseconds then return the function".  This was only necessary for Undead Blade and Slash to match the animation.

```
                            setTimeout(() => {
                                Unit.takeDamage(unit, damageDone, state.casterUnit, underworld, prediction, state);
                            }, 100)
```

In retrospect, Undead Blade is a little more complicated than most spells and I shouldn't have made it the first mod example.

TL;DR you can remove this block, it just makes the function wait 100 ms before returning.